src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -783,6 +784,34 @@ public function testFloatKeys()
 
         $this->assertEquals($expected, $this->parser->parse($yaml));
     }
+
+    /**
+     * @group legacy
+     * throw ParseException in Symfony 3.0
+     */
+    public function testColonInMappingValueException()
+    {
+        $yaml = <<<EOF
+foo: bar: baz
+EOF;
+
+        $this->deprecations = array();
+        set_error_handler(array($this, 'handleError'));
```

Comment:
Can you inline the error handler as an anonymous function? It's done like that in Twig because of PHP 5.2 but we don't need that here. I should have referenced the Twig docs instead where it's done the "modern way" (sorry about that): http://twig.sensiolabs.org/doc/recipes.html#displaying-deprecation-notices